rpcd/playbooks/roles/rpc_pre_upgrade/tasks/remove_beaver.yml

```diff
@@ -0,0 +1,90 @@
+---
+# Copyright 2016, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+- name: Make sure beaver is stopped
+  service:
+    name: beaver
+    enabled: no
+    state: stopped
+  ignore_errors: true
+  delegate_to: {{ item }}
+  with_items: {{ groups['all'] }}
+  tags:
+    - remove-beaver
+
+- name: Remove beaver etc files
+  file:
+    path: /etc/beaver
+    state: absent
+  delegate_to: {{ item }}
+  with_items: {{ groups['all'] }}
+  tags:
+    - remove-beaver
+
+- name: Remove beaver init files
+  file:
+    path: /etc/init.d/beaver
+    state: absent
+  delegate_to: {{ item }}
+  with_items: {{ groups['all'] }}
+  tags:
+    - remove-beaver
+
+- name: Remove beaver logrotate files
+  file:
+    path: /etc/logrotate.d/beaver
+    state: absent
+  delegate_to: {{ item }}
+  with_items: {{ groups['all'] }}
+  tags:
+    - remove-beaver
+
+- name: Remove beaver log files
+  file:
+    path: /var/log/beaver
+    state: absent
+  delegate_to: {{ item }}
+  with_items: {{ groups['all'] }}
+  tags:
+    - remove-beaver
+
+- name: Uninstall beaver pip package
+  pip:
+    name: beaver
+    state: absent
+  ignore_errors: true
```

Comment:
One final thought is that errors are cause for concern about the general state of the environment as it is indicative of things being added without all of the plays being completed  successfully.  I do think that we should ignore said errors but a release note stating that errors in the remove beaver role should at least warrant a sanity check.